src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb

```diff
@@ -571,6 +569,22 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
                 tList.Add(MakeWhiteSpaceTrivia(GetText(Here - 1)))
             End If
 
+            If foundComment Then
+                Dim comment As SyntaxTrivia = ScanComment()
+                If Not CheckFeatureAvailability(Feature.CommentsAfterLineContinuation) Then
+                    comment = comment.WithDiagnostics({ErrorFactory.ErrorInfo(ERRID.ERR_CommentsAfterLineContinuationNotAvailable1,
+                        New VisualBasicRequiredLanguageVersion(Feature.CommentsAfterLineContinuation.GetLanguageVersion()))})
+                End If
+                tList.Add(comment)
+                ' Need to call CanGet here to prevent Peek reading past EndOfBuffer. This can happen when file ends with comment but no New Line.
+                If CanGet() Then
+                    ch = Peek()
+                    atNewLine = IsNewLine(ch)
+                Else
+                    atNewLine = False
```

Comment:
>atNewLine = False [](start = 20, length = 17)

Isn't this assignment redundant?

Reply:
No. The comment could end with NewLine or EndOfFile. I could put the peek inline.

Reply:
> No. The comment could end with NewLine or EndOfFile. I could put the peek inline. 

Let me rephrase the question. What values can ```atNewLine``` variable have when we enter the Else block?

---
In reply to: [217802873](https://github.com/dotnet/roslyn/pull/28327#discussion_r217802873) [](ancestors = 217802873)

Reply:
I understand your point, I believe only false but I added a debug.assert until I am more comfortable with that statement.